src/Core/Services/CryptoService.cs

```diff
@@ -62,6 +62,11 @@ public Task<UserKey> GetUserKeyAsync(string userId = null)
             return _stateService.GetUserKeyAsync(userId);
         }
 
+        public async Task<bool> IsLegacyUserAsync(MasterKey masterKey = null, string userId = null)
+        {
+            return await ValidateUserKeyAsync(new UserKey((masterKey ?? await GetMasterKeyAsync(userId)).Key));
+        }
```

Comment:
⚠️  `GetMasterKeyAsync(...)` could potentially [return null](https://github.com/bitwarden/mobile/blob/efd1671f489186e529ed9ff199b99c74b7bb9a21/src/Core/Services/CryptoService.cs#L133). Is it ok if that happens to throw a null reference exception here (when accessing `.Key`)?

Reply:
Good catch, added a check